store/commitment/store.go

```diff
@@ -0,0 +1,151 @@
+package commitment
+
+import (
+	"errors"
+	"fmt"
+
+	dbm "github.com/cosmos/cosmos-db"
+	ics23 "github.com/cosmos/ics23/go"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/store/v2"
+	"cosmossdk.io/store/v2/commitment/iavl"
+	"cosmossdk.io/store/v2/commitment/types"
+)
+
+const (
+	// DefaultTreeType is the default tree type.
+	DefaultTreeType = "default"
+	IavlTreeType    = "iavl"
+)
+
+var _ store.Committer = (*CommitStore)(nil)
+
+// CommitStore is a wrapper around the state Tree.
+type CommitStore struct {
+	multiTrees map[string]types.Tree
+}
+
+// NewCommitStore creates a new CommitStore instance.
+func NewCommitStore(storeConfigs map[string]interface{}, db dbm.DB, logger log.Logger) (*CommitStore, error) {
+	multiTrees := make(map[string]types.Tree)
+	for storeKey, cfg := range storeConfigs {
+		switch config := cfg.(type) {
+		case *iavl.Config:
+			iavlDB := dbm.NewPrefixDB(db, []byte(storeKey))
+			multiTrees[storeKey] = iavl.NewIavlTree(iavlDB, logger, config)
+		default:
+			return nil, fmt.Errorf("unknown tree type for store %s, config: %v", storeKey, cfg)
+		}
+	}
```

Comment:
This is for allowing the different backends each store.

Reply:
Trying to read @tac0turtle's mind here, I think he wants to avoid direct import of IAVL and avoid casing altogether. Instead, can we just take `map[string]Tree` as an arg?

Reply:
the import seems fine, i dont understand the need for a switch statement here. Its one case, are users meant to pr here to add new tree types?

Reply:
exactly, they also should implement the Tree wrapper

Reply:
> Instead, can we just take map[string]Tree as an arg?
@bez I think this is the appropriate place to interact with the Tree constructor, I don't want to propagate iavl import to the upper side.

Reply:
we cant users to pr here everytime they want to add a new structure, this will slow down their development and force them to fork and possibly diverge faster becuase the release is blocked. Is there anyway to avoid this? 

I agree with leaving the any type here because we dont know if the tree interface will always be used for a store key or something else. 

Its still unclear to me how a user would register a different commitment structure to be used for a storekey?

Reply:
hmm, that's a good point, IMO it is not possible to avoid PR for different backend.
How can interact with backend without interface?

Reply:
in that case, i would propose this move from commitment to a subpkg in commitment called tree. Or at least some identifier that will show the reader that this is meant for trees only. At a higher level, i think it makes sense for the call to be able to set tree types in the map and then only for the unset ones would we set iavl. If nothing is set, we set the default.

Reply:
ok, I understand your point, then I agree with @alexanderbez 's point, passing `map[string]Tree` in the constructor.
When users want to add new backend, they should implement Tree interface on their side and create the PR against the baseapp or whatever the manager module. It will not block the store module release, right?

Reply:
It should not block store release, correct.